aiida/backends/sqlalchemy/tests/session.py

```diff
@@ -181,3 +180,49 @@ def test_session_update_and_expiration_4(self):
         code.store()
 
         self.drop_connection()
+
```

Comment:
I would also add a test without any SQL specific command (in this case `expunge_all` etc), as this is what the user would do (this could be a generic test) that would have triggered the error before.

Reply:
This is a good point. 

The problem is that I have not managed to find until today (and it has been a while that I investigate this) a way to isolate the problem at that level (which more or less implies to follow the DbNode object generation in @nmounet workflows).

Reply:
Ok. For now, then, at least add the same identical test (but without any sqla command) as a general test. Then for me we can merge. Thanks!

Reply:
OK, I will. 

Just keep in mind that it will be a test that doesn't test anything. I.e. that test would not fail even with the problematic code of set_value() that was corrected.

Reply:
I created the test but I did not push it yet. Let's have a short chat before I add it to this PR.

Reply:
ok, I see. Let's merge this, then.